cmd/manager/main.go

```diff
@@ -321,6 +321,9 @@ func execute() {
 			log.Error(err, "unable to create controller", "controller", "LicenseTrial")
 			os.Exit(1)
 		}
+
+		r := licensing.NewResourceReporter(mgr.GetClient())
+		go r.Start(operatorNamespace)
```

Comment:
Since we're using the manager's client we should be using its cache, but the manager also waits for the cache to be synced before starting the controllers.

I need to look at it more but I wonder if we want to run this to be managed by the Manager. It looks like we just need to implement `Start(<-chan struct{}) error`. I also think we can explicitly wait for the cache sync with `mgr.GetCache().WaitForCacheSync()` if we have to.

But that might not be necessary. For operators we really don't want to be operating on an incomplete cache, but for this I think it just means that the first run might be inaccurate. After three minutes the cache should be synced and the next run will be correct.

Reply:
Calling `WaitForCacheSync ` sounds like a good idea to me
```
go func() {
    mgr.GetCache().WaitForCacheSync()
    r := licensing.NewResourceReporter(mgr.GetClient())
    r.Start(operatorNamespace)
}()
```

Reply:
Looks cleaner to wait for the cache sync. Let's do that.